ssl/s3_lib.c

```diff
@@ -3586,8 +3586,8 @@ long ssl3_ctrl(SSL *s, int cmd, long larg, void *parg)
 
             if (!s->session)
                 return 0;
```

Comment:
It seems pretty likely that this ```if (!s->session)``` was just to avoid a NULL dereference; it may well be stale logic at this point.  (That said, it's probably harmless, since we will definitely have an attached session by the point that we're parsing the peer's group list.)

Reply:
if the line is kept, make it explicit test against NULL